docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,728 @@
+    #     ami: ami-0f46c2ed46d8157aa
+    #
+    #
+    # Example for Azure:
+    #
+    # apiVersion: azure.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 1967.5.0
+    #   publisher: CoreOS
+    #   offer: CoreOS
+    #   sku: Stable
+    # countFaultDomains:
+    # - region: westeurope
+    #   count: 2
+    # countUpdateDomains:
+    # - region: westeurope
+    #   count: 5
+    #
+    #
+    # Example for GCP:
+    #
+    # apiVersion: gcp.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: projects/coreos-cloud/global/images/coreos-stable-2023-5-0-v20190312
+    #
+    #
+    # Example for OpenStack:
+    #
+    # apiVersion: openstack.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: coreos-2023.5.0
+    # keyStoneURL: https://url-to-keystone/v3/
+    # dnsServers:
+    # - 10.10.10.10
+    # - 10.10.10.11
+    # dhcpDomain: foo.bar
+    # requestTimeout: 30s
+    # constraints:
+    #   loadBalancerProviders:
+    #   - name: haproxy
+    #   floatingPools:
+    #   - name: fip1
+    #     loadBalancerClasses:
+    #     - name: class1
+    #       floatingSubnetID: 04eed401-f85f-4610-8041-c4835c4beea6
+    #       floatingNetworkID: 23949a30-1cdd-4732-ba47-d03ced950acc
+    #       subnetID: ac46c204-9d0d-4a4c-a90d-afefe40cfc35
+    #
+    #
+    # Example for Packet:
+    #
+    # apiVersion: packet.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2079.3.0
+    #   id: d61c3912-8422-4daf-835e-854efa0062e4
+```
+
+### `Seed` resource
+
+Special note: The proposal contains fields that are not yet existing in the current `garden.sapcloud.io/v1beta1.Seed` resource, but they should be implemented (open issues that require them are linked).
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: seed-secret
+  namespace: garden
+type: Opaque
+data:
+  kubeconfig: base64(kubeconfig-for-seed-cluster)
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: Seed
+metadata:
+  name: seed1
+spec:
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    region: europe-central-1
+  secretRef:
+    name: seed-secret
+    namespace: garden
+  # Motivation for DNS section: https://github.com/gardener/gardener/issues/201.
+  dns:
+    provider: <some-provider-name> # {aws-route53, google-clouddns, ...}
+    secretName: my-dns-secret # must be in `garden` namespace
+    ingressDomain: seed1.dev.example.com
+  volume: # optional (introduced to get rid of `persistentvolume.garden.sapcloud.io/minimumSize` and `persistentvolume.garden.sapcloud.io/provider` annotations)
+    minimumSize: 20Gi
+    providers:
```

Comment:
Thank you @rfranzke  for careful campatible consideration. I don't see the need to support multiple PV providers. Maybe you are considering below case: e.g. for aws k8s in-tree volume plugin. And some day, we support CSI solution. But even though, why we need to maintain 2 providers here?

Reply:
We don't know what the provider extensions use these configurations for. You, with Alicloud, are using it for the storage class for the Alicloud seed itself. Maybe other providers want to implement different storage classes for different purposes (e.g., prometheus uses a special storageclass compared to etcd-main or etcd-events).
I don't have a strong opinion, but now that it's already implemented in #1308 I suggest to keep it as is. It doesn't harm, it just gives more flexibility, and we don't use it internally anyways.